nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/AssetKind.java

```diff
@@ -25,18 +25,35 @@
  */
 public enum AssetKind
 {
-  HELM_INDEX(CacheControllerHolder.METADATA),
-  HELM_PROVENANCE(CacheControllerHolder.CONTENT),
-  HELM_PACKAGE(CacheControllerHolder.CONTENT);
+  HELM_INDEX(CacheControllerHolder.METADATA, null),
+  HELM_PROVENANCE(CacheControllerHolder.CONTENT, ".tgz.prov"),
+  HELM_PACKAGE(CacheControllerHolder.CONTENT, ".tgz");
 
   private final CacheType cacheType;
+  private final String extension;
 
-  AssetKind(final CacheType cacheType) {
+  AssetKind(final CacheType cacheType, final String extension) {
     this.cacheType = cacheType;
+    this.extension = extension;
+  }
+
+  public static AssetKind getAssetKindByFileName(final String name) {
+    if (name.endsWith(HELM_PACKAGE.getExtension())) {
+      return AssetKind.HELM_PACKAGE;
+    }
+    else if (name.endsWith(HELM_PROVENANCE.getExtension())) {
+      return AssetKind.HELM_PROVENANCE;
+    }
+    return AssetKind.HELM_INDEX;
   }
 
   @Nonnull
   public CacheType getCacheType() {
     return cacheType;
   }
+
+  @Nonnull
+  public String getExtension() {
```

Comment:
Can be null for HELM_INDEX
Change to Nullable

Reply:
I think better define ".yaml" for HELM_INDEX